### PR TITLE
Add design to preprocess feature inputs using analysis results.

### PR DESCRIPTION
**Pull request by workingloong**

### CODE REVIEW

File: docs/designs/preprocess_with_analysis.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
+# Preprocess with Analysis Result Design
+This document describes the design of preprocessing feature inputs using analysis results.
+
+## Motivation
+Before preprocessing the feature inputs, we need to analyze the training dataset to collect the feature statistics. For example, we need the mean and standard deviation to normalize a numeric value, `vocabulary` to lookup a string value to an integer id and `boundary` to discretize a numeric value. Using SQLFlow, the training dataset is usually a table saved in MySQL or MaxCompute and other databases, so we can use SQL to analyze the training table. During [data transformation pipeline](https://github.com/sql-machine-learning/elasticdl/blob/develop/docs/designs/data_transform.md), we may launch a pod to analyze the training table and then submit the ElasticDL training job. So, the design is to solve how to pass the analysis results into the pods of an ElasticDL training job.
+
```
Comment (QiJune) [inline docs/designs/preprocess_with_analysis.md:6]: There are two steps in preprocessing: analysis and transformation. The results generated by analysis will be used in transformation.  It seems that it's a workflow.  Please refer to [here](https://github.com/sql-machine-learning/fluid/blob/develop/doc/design.md).

I am not sure if we could use workflow to describe the data processing pipeline, and also the dependencies between steps.

Comment (workingloong) [reply docs/designs/preprocess_with_analysis.md:6]: Yes, it is a pipeline. We will use workflow to describe the data preprocessing and training job when we use it in SQLFlow. The design only discusses how to pass the results generated by analysis into training pods to build the model. We can orchestrate the analysis containers and training containers using k8s workflow or PAI-GUI.

File: docs/designs/preprocess_with_analysis.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
+# Preprocess with Analysis Result Design
+This document describes the design of preprocessing feature inputs using analysis results.
```
Comment (brightcoder01) [inline docs/designs/preprocess_with_analysis.md]: This document describes the design about how to use analysis result while preprocessing feature inputs.

File: docs/designs/preprocess_with_analysis.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+## Motivation
+Before preprocessing the feature inputs, we need to analyze the training dataset to collect the feature statistics. For example, we need the mean and standard deviation to normalize a numeric value, `vocabulary` to lookup a string value to an integer id and `boundary` to discretize a numeric value. Using SQLFlow, the training dataset is usually a table saved in MySQL or MaxCompute and other databases, so we can use SQL to analyze the training table. During [data transformation pipeline](https://github.com/sql-machine-learning/elasticdl/blob/develop/docs/designs/data_transform.md), we may launch a pod to analyze the training table and then submit the ElasticDL training job. So, the design is to solve how to pass the analysis results into the pods of an ElasticDL training job.
```
Comment (brightcoder01) [inline docs/designs/preprocess_with_analysis.md]: the feature statistics -> the feature statistical results.

File: docs/designs/preprocess_with_analysis.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+## Define preprocess layers with analysis result
+
+### 1. Persist the analysis result collected in the analysis pod?
```
Comment (brightcoder01) [inline docs/designs/preprocess_with_analysis.md]: Why add "?" here

Comment (workingloong) [reply docs/designs/preprocess_with_analysis.md]: Yeah, replace it with "." .

**Review by brightcoder01 — APPROVED** [review]

**Pull request merged**